.NET/Microsoft.Recognizers.Text.DateTime/Parsers/BaseDateTimePeriodParser.cs

```diff
@@ -1097,55 +1097,60 @@ private DateTimeResolutionResult MergeTwoTimePoints(string text, DateObject refe
             DateObject pastBegin = (DateObject)((DateTimeResolutionResult)pr1.Value).PastValue,
                        pastEnd = (DateObject)((DateTimeResolutionResult)pr2.Value).PastValue;
 
-            if (bothHaveDates)
+            // If one side contains "ampm" while the other doesn't, shift the time appropriately
+            var ampmStr1 = ((DateTimeResolutionResult)pr1.Value).Comment;
+            var ampmStr2 = ((DateTimeResolutionResult)pr2.Value).Comment;
+            if (ampmStr1 is Constants.Comment_AmPm ^ ampmStr2 is Constants.Comment_AmPm)
             {
-                if (futureBegin > futureEnd)
+                if (futureBegin > futureEnd && futureBegin.Date == futureEnd.Date)
                 {
-                    futureBegin = pastBegin;
+                    futureEnd = futureEnd.AddHours(Constants.HalfDayHourCount);
                 }
 
-                if (pastEnd < pastBegin)
+                if (pastBegin > pastEnd && pastBegin.Date == pastEnd.Date)
                 {
-                    pastEnd = futureEnd;
+                    pastEnd = pastEnd.AddHours(Constants.HalfDayHourCount);
                 }
             }
 
+            var leftTimex = pr1.TimexStr;
+            var rightTimex = pr2.TimexStr;
+
             if (bothHaveDates)
             {
-                var duration = futureEnd - futureBegin;
-                var durationStr = Convert.ToInt32(duration.TotalHours) != 0 ? $"{Convert.ToInt32(duration.TotalHours)}H" :
-                    $"{Convert.ToInt32(duration.TotalMinutes)}M";
-                ret.Timex = $"({pr1.TimexStr},{pr2.TimexStr},PT{durationStr})";
+                if (futureBegin > futureEnd)
+                {
+                    futureBegin = pastBegin;
+                }
 
-                // Do nothing
+                if (pastEnd < pastBegin)
+                {
+                    pastEnd = futureEnd;
+                }
             }
             else if (beginHasDate)
             {
                 futureEnd = DateObject.MinValue.SafeCreateFromValue(
                     futureBegin.Year, futureBegin.Month, futureBegin.Day, futureEnd.Hour, futureEnd.Minute, futureEnd.Second);
-
                 pastEnd = DateObject.MinValue.SafeCreateFromValue(
                     pastBegin.Year, pastBegin.Month, pastBegin.Day, pastEnd.Hour, pastEnd.Minute, pastEnd.Second);
 
-                var dateStr = pr1.TimexStr.Split('T')[0];
-                var durationStr = DateTimeFormatUtil.LuisTimeSpan(futureEnd - futureBegin);
-                ret.Timex = $"({pr1.TimexStr},{dateStr + pr2.TimexStr},{durationStr})";
+                rightTimex = pr1.TimexStr.Equals(Constants.TimexNow, StringComparison.Ordinal) ? DateTimeFormatUtil.LuisDateShortTime(futureEnd) :
```

Comment:
Couldn't these timex logic be turned into a parametrized method in TimexUtils?
Please add it in a follow up PR.

Reply:
done